R/util.R

```diff
@@ -1475,39 +1475,49 @@ one_hot <- function(df, key) {
   df %>% tidyr::spread(!!rlang::enquo(key), !!rlang::sym(tmp_value_col), fill = 0, sep = "_") %>% select(-!!rlang::sym(tmp_id_col))
 }
 
+# API to get a list of argument names
+extract_argument_names <- function(...) {
+  q <- rlang::quos(...)
+  purrr::map(q, function(x){rlang::quo_name(x)})
+}
 
 #'Wrapper function for dplyr::bind_rows to support named data frames when it's called inside dplyr chain.
 #'@export
-bind_rows <- function(..., .id = NULL, first_id = '', ignore_column_data_type = FALSE) {
+bind_rows <- function(..., id_column_name = NULL, current_df_name = '', force_data_type = FALSE) {
+  # get a list of argument names to resolve data frame names passed to this bind_rows.
+  # only exception is the current data frame which is passed via dplyr pipe operation (%>%).
+  # it becomes period (.) instead of actual df name.
+  args <- extract_argument_names(...)
+  browser()
   # If the dplyr::bind_rows is called within a dplyr chain like df1 %>% dplyr::bind_rows(list(df_2 = df2, df_3 = df3), .id="id"),
   # since df1 does not have a name, the "id" column of the resulting data frame does not have the data frame name for rows from df1.
   # To workaround this issue, set a name to the first data frame with the value specified by fistLabel argument as a pre-process
   # then pass the updated list to dplyr::bind_rows.
   dataframes_updated <- list()
   # with dplyr:::flatten_bindable API, create a list of data frames from arguments passed to bind_rows.
   dataframes <- dplyr:::flatten_bindable(rlang::dots_values(...))
-  if(ignore_column_data_type | stringr::str_length(first_id) >0) {
+  if(force_data_type | stringr::str_length(current_df_name) >0) {
     index <- 1;
     # for the case where a user passes a list that contains key (data frame name) and value (data frame) pair.
     if(!is.null(names(dataframes))) {
       # iterate data frames list by name as a key.
       for (name in names(dataframes)) {
         # for the first item, it's the data frame passed via %>% operator, so it does not have a key (data frame name) yet.
         # so populate the key with the value passed by first_id argument.
-        if(stringr::str_length(first_id) > 0 & index == 1) {
-          # if ignore_column_data_type is set, force character as column data types
-          if(ignore_column_data_type) {
-            dataframes_updated[[first_id]] <- dplyr::mutate_all(dataframes[[1]], funs(as.character))
+        if(stringr::str_length(current_df_name) > 0 & index == 1) {
```

Comment:
Use && instead of & in if statement condition to make sure the resulting logical vector is of length 1.

Reply:
Fixed as suggested